docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,190 @@
+
+[NOTE]
+You should not use placeholders for configuration options that can only be set through the `build` command. The reason is that these options
+can not change once they are persisted into the server image. Prefer using placeholders only for the configurations options available to
+the `start` command.
+
+== Configuration profiles
+
+Most of the time the server is deployed to different environments like testing, staging, and production environments. Although they share some
+core configuration options (e.g: database, clustering, etc) some other options might change on a per-environment basis.
+In order to make easier to support multiple configurations, the server allows you to create configuration profiles.
+
+Configuration profiles are basically a set of one or more configuration options grouped together using a unique key.
+
+You can activate a profile by using the `[-pf|--profile]` option as follows:
+
+<@kc.all rootParameters="-pf myprofile"/>
+
+Once a profile is activated, all the configuration options defined for the profile should take effect when configuring or starting the server.
+
+=== Understanding the default profiles
+
+By default, the server defines two main profiles:
+
+* `dev`
+* `prod`
+
+The `dev` profile is activated everytime you run the `start-dev` command. In this profile, some key configuration options are set to make possible to start the
+server for development purposes without the burden of having to define additional settings that are mandatory for production.
+
+The `prod` profile is activated by default whenever you run the `build` or the `start` command. You should use this profile to set any configuration option that
+makes sense when deploying the server in production.
+
+By default, the configurations options for the `prod` profile are commented in the `conf/keycloak.properties`. By looking at these options
+you should have an idea about the main settings that you should consider when running in production.
+
+You should be able to change the configuration options for these built-in profiles as well as create your own.
+
+=== Defining a configuration profile
+
+A profile can be created by adding a set of one or more properties to the `conf/keycloak.properties` file as follows:
+
+```
+%myprofile.db.username=foo
+%myprofile.db.password=foo_secret
+%myprofile.hostname=mykeycloak
+```
+
+Any configuration option associated with a profile should be prefixed by a `%` followed by the profile name and a dot `.` like in the example above.
+
+== Using Quarkus configuration options
+
+The server is built on top of <@link.external url="https://quarkus.io" text="Quarkus"/> and you should be able to use any property available from Quarkus
+if you do not find a corresponding options in the server configuration. The server has a configuration layer that provides a 1:N mapping between the server configuration options and those available from Quarkus.
```

Comment:
Imo this is too detailed for a user guide. Imo the important thing here is: "I want to know how I can use quarkus properties", not "what is our configuration layer doing internally".

Reply:
So a short example, together with the statement that we'd prefer to have a discussion on why ppl need it and add it to our layer would be better i think.

Reply:
Yeah, kind of exposing a bit of details. But the idea is to express that we do support Quarkus properties and if you are doing that is because we can probably improve our configuration.

I do not see an example helping much.

Or perhaps we should remove this entirely and move to some other guide?

Reply:
+1 for removing completely, for now. Only generates noise imo, as is. :)

Alternatively, I thought about sth like this:

```
== Using Quarkus configuration options
The server is built on top of <@link.external url="https://quarkus.io" text="Quarkus"/> and you are basically able to use any property available within Quarkus, using for example `quarkus.datasource.jdbc.max-size=...`. 

We generally want to provide an extra convenience layer over the raw Quarkus properties using our own configuration. So we'd love to hear from you over at <@link.external url="https://github.com/keycloak/keycloak/discussions" text="GitHub Discussions"/>. if you think something is missing.
``` 


Either way fine for me ;)